test/Test/Database/LSMTree/Internal.hs

```diff
@@ -207,13 +206,45 @@ prop_roundtripCursor kops = ioProperty $
     withTempIOHasBlockIO "prop_roundtripCursor" $ \hfs hbio -> do
       withSession nullTracer hfs hbio (FS.mkFsPath []) $ \sesh -> do
         withTable sesh conf $ \th -> do
-          updates appendSerialisedValue (coerce kops) th
-          withCursor th $ \_cursor ->
-            -- TODO: read from cursor once implemented!
-            return ()
+          updates appendSerialisedValue (coerce kopsWithoutblobs) th
+          fromCursor <- withCursor th $ readWholeCursor appendSerialisedValue
+          return $ expected === fromCursor
   where
     conf = testTableConfig
 
+    -- TODO: include inserts with blobs once write buffer returns BlobRefs
+    kopsWithoutblobs = V.map (fmap f) kops
+      where f (InsertWithBlob v _) = Insert v
+            f x                    = x
+
+    expected =
+      V.fromList . mapMaybe (traverse entryToValue) $
+          Map.assocs . Map.fromListWith (combine appendSerialisedValue) $
```

Comment:
Will this test ever encounter duplicates? It might be worth labelling so that we can see

Reply:
I added labelling, or rather tabulation to also see how many keys in a test run are duplicates and how often they occur.

```
        +++ OK, passed 100 tests.
        
        duplicates (46 in total):
        85% 2
        13% 3
         2% 7
```

So not often, but they exist.